### ISSUE TITLE
Flexible 'private options'

### ISSUE THREAD

**Issue by fran-worley** [apotonick/formular#4]
Because of the way that this works:
https://github.com/apotonick/formular/blob/master/lib/formular/builder.rb#L113-L120

We are unable to add options specific to a front end without adding to the list of supported options in Formular Base. Already I have added :hint, :column_attrs and :style for bootstrap cases. I appreciate that some of these might be useful to other frameworks but it strikes me that that Formlar::Builder is not the best place to define front-end options.

If there was a way to define front end specific options in the front end class then there would be no need to update Formular itself each time.

I am raising this largely because I want to add support for input groups which will require a number of custom options and I don't want to have to include these onto Formular::Builder private_options.

Happy to put together a PR but wanted to see what your thoughts were on the subject.

**Comment by apotonick** [comment]
I thought about this, too, and it's kinda simple to separate that into the concrete frontend class, but so far, I didn't see any reason to "over-engineer" :giggle: - what options are you thinking of adding?

**Comment by fran-worley** [comment]
I had been looking at something like this: 

1 option called `:input groups`
```ruby
f.input :email, label: "Email", input_groups: { 
  left: { 
    content: "@", 
    btn: false, # false will be the default, required as if true the input-group needs a different class
    attrs: { class: ["some-class"], id: "left-group", data: {} }
  }, 
  right: { 
    content: "<input type='checkbox'>", 
    attrs: { class: ["some-other-class"], id: "right-group", data: {} } 
  }
}
```
or 2 options: `:input_group_left` and `:input_group_right` to reduce hash nesting
```ruby
f.input :email, label: "Email", 
  input_group_left: { 
    content: "@", 
    btn: false, # false will be the default, required as if true the input-group needs a different class
    attrs: { class: ["some-class"], id: "left-group", data: {} } 
  }, input_group_right: { 
    content: "<input type='checkbox'>", 
    attrs: { class: ["some-other-class"], id: "right-group", data: {} } 
  }
```
Either would give you something like this:
```html
<div class="form-group">
  <label class="control-label">Email Address</label><!-- regular html control label from formular -->
  <div class="input-group">
      <span class="input-group-addon some-class" id="left-group">@</span>
      <input type="text" class="form-control" ><!-- regular html input from formular -->
      <span class="input-group-addon some-other-class" id="right-group">
        <input type="checkbox">
      </span>
    </div>
</div>
```

**Comment by fran-worley** [comment]
It looks like Foundation has something similar but they call them 'Pre/Postfix Labels & Actions'
http://foundation.zurb.com/sites/docs/v/5.5.3/components/forms.html

**Comment by apotonick** [comment]
Ok! Let's keep going with adding this to the global `Builder` for now until I got some time to look how to extract that to the respective frontend class - unless your PR is sitting and waiting to get merged, already!!!

**Comment by fran-worley** [comment]
See PR #5 I'm pretty sure there is a better way but for now I just overwrote private options and appended the bootstrap specific bits.

**Comment by apotonick** [comment]
I merged your #5 - it's great what you did there.
